scripts/docs-deploy/deploy-to-site.ts

```diff
@@ -0,0 +1,57 @@
+import {$} from 'zx';
+import {SiteTarget} from './utils';
+
+interface Deployment {
+  projectId: string;
+  site: SiteTarget;
+}
+
+/** Interface describing a production deployment. */
+export interface ProductionDeployment extends Deployment {
+  description: string;
+}
+
+/** Interface describing a temporary preview deployment. */
+export interface PreviewDeployment extends Deployment {
+  channelId: string;
+  expires: string;
+}
+
+/** Type describing a Firebase deployment. */
+export type DeploymentInfo = ProductionDeployment | PreviewDeployment;
+
+/**
+ * Deploys the docs site at the specified directory to Firebase with respect
+ * to the deployment information provided.
+ *
+ * The deployment info either describes the production deployment information,
+ * or a preview temporary deployment that will expire automatically.
+ */
+export async function deployToSite(projectPath: string, token: string, info: DeploymentInfo) {
+  // Note: Running yarn in silent move to avoid printing of tokens.
+  const firebase = async (...cmd: string[]) =>
+    await $`yarn -s --cwd ${projectPath} firebase --non-interactive --token "${token}" ${cmd}`;
```

Comment:
We should use `FIREBASE_TOKEN` to provide the token.

Reply:
Actually going through this, I think we'd want to keep the token being passed around, so that we can have different environment variables if we'd have different firebase tokens. e.g. for dev-app deployment etc.

Reply:
If I understand it correctly, we should be able to create ci credentials which allow access to multiple firebase projects.

Reply:
Yes, I think the credentials are user-coupled but not actual service accounts (that's why it might work accross projects). I believe it might be good keeping that functionality and we could always just set the token in the config to `process.env.FIREBASE_TOKEN`.